Classes/Controller/AbstractController.php

```diff
@@ -137,15 +137,12 @@ protected function loadDocument(int $documentId = 0): void
             } else if (GeneralUtility::isValidUrl($documentId)) {
 
                 $doc = AbstractDocument::getInstance($documentId, $this->settings, true);
-
                 if ($doc !== null) {
                     if ($doc->recordId) {
-                        $this->document = $this->documentRepository->findOneByRecordId($doc->recordId);
-                    }
-
-                    if ($this->document === null) {
-                        // create new dummy Document object
-                        $this->document = GeneralUtility::makeInstance(Document::class);
+                        if ($this->documentRepository->findOneByRecordId($doc->recordId) === null) {
```

Comment:
There is an `else` statement missing for when there is a valid document in the repository.

(The original code sets `$this->document` to the document retrieved from the repository and uses a dummy document only as a fallback.)

Reply:
Yes, forgot the else statement. Ill add it, but than leave the nested if statements:

```
                    if ($doc->recordId) {
                        if ($this->documentRepository->findOneByRecordId($doc->recordId) === null) {
                            // create new dummy Document object
                            $this->document = GeneralUtility::makeInstance(Document::class);
                        } else {
                            // find document from repository by recordId
                            $this->document = $this->documentRepository->findOneByRecordId($doc->recordId);
                        }
                    }
```

Reply:
You should avoid requesting the repository twice:
 ```
if ($doc->recordId) {
    // find document from repository by recordId
    $docFromRepository = $this->documentRepository->findOneByRecordId($doc->recordId);
    if ($docFromRepository !== null) {
        $this->document = $docFromRepository;
    } else {
        // create new dummy Document object
        $this->document = GeneralUtility::makeInstance(Document::class);
    }
}
```

Reply:
You're right